tests/vm_test.sh

```diff
@@ -166,4 +166,56 @@ function test_vm_umount()
   }
 }
 
+function test_vm_parse_options()
+{
+  unset options_values
+  declare -gA options_values
+  local output
+  local option_output
+
+  # test individual options
+  option_output="$(vm_help)"
+  unset options_values
+  declare -gA options_values
+  output="$(parse_vm_options -h)"
+  assertEquals "($LINENO)" "$option_output" "$output"
+
+  function vm_mount()
+  {
+    echo 'mount'
+  }
```

Comment:
If possible avoid creating functions inside functions, if you want to test what command is being executed you can use `cmd_manager` to  print the command instead of executing it.